src/main/java/de/srendi/advancedperipherals/common/blocks/blockentities/RsBridgeEntity.java

```diff
@@ -0,0 +1,129 @@
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class RsBridgeEntity extends PeripheralBlockEntity<RSBridgePeripheral> implements IPeripheralTileEntity, NetworkNodeContainerProvider, TaskStatusListener {
+
+    protected CompoundTag peripheralSettings;
+    private final NetworkNode node;
+    private final InWorldNetworkNodeContainer networkNodeContainer;
+    private final List<RSCraftJob> jobs = new CopyOnWriteArrayList<>();
+    private boolean addedListener = false;
+
+    public RsBridgeEntity(BlockPos pos, BlockState state) {
+        super(BlockEntityTypes.RS_BRIDGE.get(), pos, state);
+        peripheralSettings = new CompoundTag();
+        ConnectionStrategy connectionStrategy = new SimpleConnectionStrategy(pos);
+        node = new SimpleNetworkNode(APConfig.PERIPHERALS_CONFIG.rsConsumption.get());
+        networkNodeContainer = new InWorldNetworkNodeContainerImpl(this, node, "RS Bridge", 1, connectionStrategy, null);
+    }
+
+    @NotNull
+    @Override
+    protected RSBridgePeripheral createPeripheral() {
+        return new RSBridgePeripheral(this);
+    }
+
+    @Override
+    public CompoundTag getPeripheralSettings() {
+        return peripheralSettings;
+    }
+
+    @Override
+    public void markSettingsChanged() {
+        setChanged();
+    }
+
+    @Override
+    public <T extends BlockEntity> void handleTick(Level level, BlockState state, BlockEntityType<T> type) {
+        if (getNode().getNetwork() != null) {
+            AutocraftingNetworkComponent manager = getNode().getNetwork().getComponent(AutocraftingNetworkComponent.class);
+            if (!this.addedListener) {
+                manager.addListener(this);
+                this.addedListener = true;
+            }
+        }
+
+        // Try to start the job if the job calculation finished
+        jobs.forEach(BasicCraftJob::tick);
+
+        // Remove the job if the crafting calculation failed, we can't do anything with it anymore
+        jobs.removeIf(BasicCraftJob::canBePurged);
+    }
+
+    public void addJob(RSCraftJob job) {
+        jobs.add(job);
+    }
+
+    public List<RSCraftJob> getJobs() {
+        return jobs;
+    }
+
+    public NetworkNode getNode() {
+        return node;
+    }
+
+    public void clearRemoved() {
+        super.clearRemoved();
+        initialize(this.level, null);
+    }
+
+    @NotNull
+    @Override
+    public Set<InWorldNetworkNodeContainer> getContainers() {
+        return Set.of(networkNodeContainer);
+    }
+
+    @Override
+    public void addContainer(@NotNull InWorldNetworkNodeContainer inWorldNetworkNodeContainer) {
+    }
+
+    @Override
+    public boolean canBuild(@NotNull ServerPlayer serverPlayer) {
+        return true;
+    }
+
+    @Override
+    public void taskStatusChanged(@NotNull TaskStatus taskStatus) {
+        jobs.stream().filter(job -> job.isCraftingStarted() && job.getCraftingTask().info().id().equals(taskStatus.info().id())).forEach(BasicCraftJob::jobStateChanged);
```

Comment:
instead of use `&&`, we can chain up multiple filters, which may looks cleaner.

```suggestion
        jobs.stream().filter(BasicCraftJob::isCraftingStarted).filter(job -> job.getCraftingTask().info().id().equals(taskStatus.info().id())).forEach(BasicCraftJob::jobStateChanged);
```

same for `taskRemoved` and `taskAdded`